bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeBinding.java

```diff
@@ -369,28 +339,19 @@ private void readEcobee(OAuthCredentials oauthCredentials, Selection selection)
 		// Iterate through bindings and update all inbound values.
 		for (final EcobeeBindingProvider provider : this.providers) {
 			for (final String itemName : provider.getItemNames()) {
-				if (provider.isInBound(itemName)) {
+				if (provider.isInBound(itemName)
+						&& oauthCredentials == getOAuthCredentials(provider.getUserid(itemName))) {
```

Comment:
i am pretty sure you didn't want to check for object identity here, right? Better use equals() …